testIntegration/tests/userAgent.test.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+"use strict";
+import fs from "fs-extra";
+import nock from "nock";
+import { expect } from "chai";
+import { ShopperCustomers } from "commerce-sdk/dist/customer/customer";
+
+// Helper to make the user agent being tested a bit more obvious
+async function mockRequestWithUserAgent(userAgent: string): Promise<void> {
+  nock("http://somewhere")
+    .matchHeader("user-agent", userAgent)
```

Comment:
add tests for client specific user agents are not overwritten

Reply:
Added.